File: packages/shared-types/src/dtos/time-logs/time-log.dto.ts
import { EntityTimestamps } from '../../types/common.types';

/**
 * Time log DTO
 */
export interface TimeLogDto extends EntityTimestamps {
  id: string;
  taskId: string | null;
  projectId: string | null;
  startTime: string;  // ISO timestamp
  endTime: string | null;  // ISO timestamp
  duration: number | null;  // minutes
  description: string | null;
}

/**
 * Time log with context
 */
export interface TimeLogDetailDto extends TimeLogDto {
  taskTitle: string | null;
  projectName: string | null;
}

/**
 * Time log summary by project
 */
export interface TimeLogSummaryDto {
  projectId: string;
  projectName: string;
  totalMinutes: number;
  entryCount: number;
}

/**
 * Create time log request
 */
export interface TimeLogCreateRequestDto {
  taskId?: string;
  projectId?: string;
  startTime: string;
  endTime?: string;
  description?: string;
}

/**
 * Update time log request
 */
export interface TimeLogUpdateRequestDto {
  taskId?: string;
  projectId?: string;
  startTime?: string;
  endTime?: string;
  description?: string;
}

File: apps/tui/internal/application/dto/time_log_dto.go
package dto

type TimeLogDto struct {
	ID          string  `json:"id"`
	TaskID      *string `json:"taskId"`
	ProjectID   *string `json:"projectId"`
	StartTime   string  `json:"startTime"`
	EndTime     *string `json:"endTime"`
	Duration    *int    `json:"duration"`
	Description *string `json:"description"`
	CreatedAt   string  `json:"createdAt"`
	UpdatedAt   string  `json:"updatedAt"`
}

type TimeLogCreateRequest struct {
	TaskID      *string `json:"taskId,omitempty"`
	ProjectID   *string `json:"projectId,omitempty"`
	StartTime   string  `json:"startTime"`
	EndTime     *string `json:"endTime,omitempty"`
	Duration    *int    `json:"duration,omitempty"`
	Description *string `json:"description,omitempty"`
}
